packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -468,414 +425,314 @@ class _MinuteControl extends StatelessWidget {
-            child: Text(formattedMinute, style: minuteStyle, textAlign: TextAlign.start, textScaleFactor: 1.0),
-          ),
-        ),
+      child: _HourMinuteControl(
+        isSelected: fragmentContext.mode == _TimePickerMode.minute,
+        text: formattedMinute,
+        onTap: Feedback.wrapForTap(() => fragmentContext.onModeChange(_TimePickerMode.minute), context),
       ),
     );
   }
 }
 
-/// Provides time picker header layout configuration for the given
-/// [timeOfDayFormat] passing [context] to each widget in the
-/// configuration.
-///
-/// The [timeOfDayFormat] and [context] arguments must not be null.
-_TimePickerHeaderFormat _buildHeaderFormat(
-  TimeOfDayFormat timeOfDayFormat,
-  _TimePickerFragmentContext context,
-  Orientation orientation,
-) {
-  // Creates an hour fragment.
-  _TimePickerHeaderFragment hour() {
-    return _TimePickerHeaderFragment(
-      layoutId: _TimePickerHeaderId.hour,
-      widget: _HourControl(fragmentContext: context),
-    );
-  }
 
-  // Creates a minute fragment.
-  _TimePickerHeaderFragment minute() {
-    return _TimePickerHeaderFragment(
-      layoutId: _TimePickerHeaderId.minute,
-      widget: _MinuteControl(fragmentContext: context),
-    );
-  }
+/// Displays the am/pm fragment and provides controls for switching between am
+/// and pm.
+class _DayPeriodControl extends StatelessWidget {
+  const _DayPeriodControl({
+    @required this.selectedTime,
+    @required this.onChanged,
+    @required this.orientation,
+  });
 
-  // Creates a string fragment.
-  _TimePickerHeaderFragment string(_TimePickerHeaderId layoutId, String value) {
-    return _TimePickerHeaderFragment(
-      layoutId: layoutId,
-      widget: _StringFragment(
-        fragmentContext: context,
-        value: value,
-      ),
-    );
-  }
+  final TimeOfDay selectedTime;
+  final Orientation orientation;
+  final ValueChanged<TimeOfDay> onChanged;
 
-  // Creates an am/pm fragment.
-  _TimePickerHeaderFragment dayPeriod() {
-    return _TimePickerHeaderFragment(
-      layoutId: _TimePickerHeaderId.period,
-      widget: _DayPeriodControl(fragmentContext: context, orientation: orientation),
-    );
+  void _togglePeriod() {
+    final int newHour = (selectedTime.hour + TimeOfDay.hoursPerPeriod) % TimeOfDay.hoursPerDay;
+    final TimeOfDay newTime = selectedTime.replacing(hour: newHour);
+    onChanged(newTime);
   }
 
-  // Convenience function for creating a time header format with up to two pieces.
-  _TimePickerHeaderFormat format(
-    _TimePickerHeaderPiece piece1, [
-    _TimePickerHeaderPiece piece2,
-  ]) {
-    final List<_TimePickerHeaderPiece> pieces = <_TimePickerHeaderPiece>[];
-    switch (context.textDirection) {
-      case TextDirection.ltr:
-        pieces.add(piece1);
-        if (piece2 != null)
-          pieces.add(piece2);
+  void _setAm(BuildContext context) {
+    if (selectedTime.period == DayPeriod.am) {
+      return;
```

Comment:
if (...) return;

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-single-line-for-short-collection-if-and-collection-for